src/main/java/nl/praegus/fitnesse/responders/symbolicLink/SymbolicLinkResponder.java

```diff
@@ -0,0 +1,124 @@
+        SimpleResponse response = new SimpleResponse();
+        response.setMaxAge(0);
+        response.setStatus(200);
+        response.setContentType("text/html");
+        response.setContent(createHTMLPage(fitNesseContext, request, wikiPage));
+
+        return response;
+    }
+
+    @Override
+    public SecureOperation getSecureOperation() {
+        return new SecureReadOperation();
+    }
+
+    public JSONArray getSymlink(WikiPage page){
+        return getSymlinkHelper(page, new JSONArray());
+    }
+
+    private JSONArray getSymlinkHelper(WikiPage wikiPage, JSONArray symbolicLinkArray) {
+        pageData = wikiPage.getData();
+        // Get Symlinks of the wiki page
+        WikiPageProperty symLinksProperty = pageData.getProperties().getProperty(SymbolicPage.PROPERTY_NAME);
+
+        // Loop through the names
+        if (symLinksProperty != null) {
+            for (String name : symLinksProperty.keySet()) {
+                symbolicLinkArray.put(getObject(name, wikiPage, symLinksProperty));
+            }
+        }
+
+        // Get Symlinks from the children
+        if (getPageChildren(wikiPage).size() > 0) {
+            for (WikiPage p : getPageChildren(wikiPage)) {
+                getSymlinkHelper(p, symbolicLinkArray);
+            }
+        }
+
+        return symLinksProperty != null ? symbolicLinkArray : new JSONArray();
+    }
+
+    private JSONObject getObject(String name, WikiPage wikiPage, WikiPageProperty symLinksProperty) {
+        JSONObject symboliclinkInformation = new JSONObject();
+        symboliclinkInformation.put("pagePath", wikiPage.getFullPath().toString());
+        symboliclinkInformation.put("linkName", name);
+        symboliclinkInformation.put("linkPath", makePathForSymbolicLink(wikiPage, symLinksProperty.get(name)));
+        symboliclinkInformation.put("backUpLinkPath", symLinksProperty.get(name));
+        return symboliclinkInformation;
+    }
+
+    private String makePathForSymbolicLink(WikiPage page, String linkPath) {
+        WikiPagePath wikiPagePath = PathParser.parse(linkPath);
+
+        if (wikiPagePath != null) {
+            WikiPage parent = wikiPagePath.isRelativePath() ? page.getParent() : page;
+            PageCrawler crawler = parent.getPageCrawler();
+            WikiPage target = crawler.getPage(wikiPagePath);
+            WikiPagePath fullPath;
+            if (target != null) {
+                fullPath = target.getFullPath();
+                fullPath.makeAbsolute();
+            } else {
+                fullPath = new WikiPagePath();
+            }
+            String pathString = fullPath.toString();
+
+            return pathString.length() > 0 && pathString.charAt(0) == '.' ? pathString.substring(1) : pathString;
+        }
+        return null;
+    }
+
+    private String createHTMLPage(FitNesseContext fitNesseContext, Request request, WikiPage wikiPage) {
```

Comment:
getHtmlPageAsString. Dat is wat je hier teruggeeft.

Reply:
Naam aangepast